### PR TITLE
Format and CI

### PR DESCRIPTION
**Pull request by Dale-Koenig**
Set up clang format to run in ci and format all files.  If this is merged and you have in progress branches, it is recommended to take the `.clang-format` file from here, run `format.sh` to format all files, commit, and THEN merge rr_devel into your branch

### CODE REVIEW

File: check_clang_format.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+echo "Showing changes in code style:"
+git diff-index HEAD
+
+# Make sure no changes have occured in repo
+if ! git diff-index --quiet HEAD --; then
+    # changes
+    echo "clang-format test failed: changes required to comply to formatting rules. See diff above.";
+    exit 1 # error
+fi
```
Comment (david-simoes-93) [inline check_clang_format.sh:39]: :+1:

File: format_cpp.sh [hunk 2]
```diff
@@ -8,4 +8,4 @@ find . \
     -not \( -path "*/build/*" -prune \) \
     -not \( -path "*/install/*" -prune \) \
     \( -name *.h -o -name *.hpp -o -name *.c -o -name *.cc -o -name *.cpp \) \
-    | xargs clang-format-5.0 -style=file -i
+    | xargs clang-format-6.0 -style=file -i
```
Comment (david-simoes-93) [inline format_cpp.sh]: in the future, we might change this to also add [a newline at the end of all files](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline), but let's leave it for another PR

**Review by david-simoes-93 — APPROVED** [review]
LGTM

**Comment by Dale-Koenig** [comment]
I will switch to clang format 10 before merging

@veerajsk waiting on approval

**Review by veerajsk — APPROVED** [review]

**Pull request merged**